rust/core-lib/src/editor.rs

```diff
@@ -998,6 +998,17 @@ impl<W: Write + Send + 'static> Editor<W> {
     // Note: the following are placeholders for prototyping, and are not intended to
     // deal with asynchrony or be efficient.
 
+    /// Applies an async edit from a plugin.
+    pub fn plugin_edit(&mut self, edit: &PluginEdit) {
+        self.this_edit_type = EditType::Other;
+        let interval = Interval::new_closed_open(edit.start as usize, edit.end as usize);
+        let text = Rope::from(&edit.text);
+        let rev_len = self.engine.get_rev(edit.rev).unwrap().len();
+        let delta = Delta::simple_edit(interval, text, rev_len);
```

Comment:
@trishume totally. The plan for a while has been to just put Deltas on the wire; I have the serde implementations for delta done in another working branch, but I've been slightly putting off finishing that patch because it requires changing both plugin libs, and i want to figure out what other changes are coming to the update protocol (we've been talking about including line break info/deltas) to avoid redundant work. But yea, I can get that work in soon.